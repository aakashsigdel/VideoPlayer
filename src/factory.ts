export const getSong = (suffix: string = '1') => ({
  title: `testSong${suffix}`,
  artist: `testArtist${suffix}`,
  videoUrl: `testUrl${suffix}`,
})

export const getSongs = () => {
  return ['1', '2', '3'].map(getSong)
}
