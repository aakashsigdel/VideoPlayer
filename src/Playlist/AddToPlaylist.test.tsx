import * as React from 'react'
import { create } from 'react-test-renderer'
import { shallow, ShallowWrapper } from 'enzyme'

import {
  AddToPlaylist,
  IAddToPlaylistProps,
  IAddToPlaylistState,
} from './AddToPlaylist'
import { getVideo } from '../factory'

describe('<AddToPlaylist />', () => {
  let wrapper: ShallowWrapper<IAddToPlaylistProps, IAddToPlaylistState>
  const defaultProps: IAddToPlaylistProps = {
    onClickAdd: jest.fn(),
    onClickCancel: jest.fn(),
  }

  beforeEach(() => {
    wrapper = shallow(<AddToPlaylist {...defaultProps} />)
  })

  it('should render correctly', () => {
    const tree = create(<AddToPlaylist {...defaultProps} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should call onClickCancel on clicking cancel button', () => {
    wrapper.find('button[type="button"]').simulate('click')
    expect(defaultProps.onClickCancel).toHaveBeenCalledTimes(1)
  })

  it('should call onClickAdd on form submit', () => {
    const video = getVideo()
    wrapper.setState(video)
    wrapper.find('form').simulate('submit', { preventDefault: jest.fn() })
    expect(defaultProps.onClickAdd).toHaveBeenCalledWith(video)
  })

  it('should set title state when typed on title input', () => {
    wrapper.find('#title').simulate('change', { target: { value: 'test' } })
    expect(wrapper.state().title).toBe('test')
  })

  it('should set artist state when typed on artist input', () => {
    wrapper.find('#artist').simulate('change', { target: { value: 'test' } })
    expect(wrapper.state().artist).toBe('test')
  })

  it('should set videoUrl state when typed on videoUrl input', () => {
    wrapper.find('#videoUrl').simulate('change', { target: { value: 'test' } })
    expect(wrapper.state().videoUrl).toBe('test')
  })
})
