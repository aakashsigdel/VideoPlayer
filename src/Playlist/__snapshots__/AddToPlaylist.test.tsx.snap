// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddToPlaylist /> should render correctly 1`] = `
<section>
  <header>
    <h2>
      Add to Playlist
    </h2>
  </header>
  <form
    onSubmit={[Function]}
  >
    <label
      htmlFor="artist"
    >
      Artist
    </label>
    <input
      id="artist"
      onChange={[Function]}
      required={true}
      type="text"
    />
    <label
      htmlFor="title"
    >
      Title
    </label>
    <input
      id="title"
      onChange={[Function]}
      required={true}
      type="text"
    />
    <label
      htmlFor="videoUrl"
    >
      Video URL
    </label>
    <input
      id="videoUrl"
      onChange={[Function]}
      required={true}
      type="url"
    />
    <footer>
      <button
        onClick={[MockFunction]}
        type="button"
      >
        Cancel
      </button>
      <button
        type="submit"
      >
        Add
      </button>
    </footer>
  </form>
</section>
`;
